src/pages/dashboard/index.tsx

```diff
@@ -19,51 +27,80 @@ interface OverviewProps {
   children?: ReactNode;
   isLoading: boolean;
 }
+interface PeriodRangeOptions {
+  pastNUnitsPeriod?: string;
+  earningPeriod?: EarningsPeriod;
+  dateRange?: { startDate: Date; endDate: Date };
+}
+
+const inputClass = `rounded p-2 bg-slate-900 text-slate-100 text-center`;
 
 const UPTIME_REQ_DOCS = "https://docs.saturn.tech/nodes-uptime-requirement";
 export const PAYOUT_STATUS_MAPPING: Record<string, string> = {
   valid: "Eligible",
   pending: "In Progress",
   postponed: "Postponed",
 };
+const earningOptions: EarningsPeriod[] = getEaringsPeriodOptions(new Date("November 1, 2022"));
 
-function createChartProps(period: TimePeriod) {
-  let dateRange;
+function createChartProps(periodType: string, periodRange: PeriodRangeOptions) {
+  let dateRange: DateRange;
   let step: DurationUnitType; // group by this time unit
   let labelUnit: DurationUnitType;
 
-  switch (period) {
-    case TimePeriod.ONE_YEAR:
-      dateRange = pastDateRange("day", 365);
-      step = "day";
-      labelUnit = "day";
+  switch (periodType) {
+    case FilterType.PastNUnits:
+      switch (periodRange.pastNUnitsPeriod) {
+        case PastNUnitsPeriod.DAY:
+          dateRange = pastDateRange("day");
+          step = "hour";
+          labelUnit = step;
+          break;
+        case PastNUnitsPeriod.MONTH:
+          dateRange = pastDateRange("day", 30);
+          step = "day";
+          labelUnit = "day";
+          break;
+        case PastNUnitsPeriod.THREE_MONTHS:
+          dateRange = pastDateRange("day", 90);
+          step = "day";
+          labelUnit = "day";
+          break;
+        case PastNUnitsPeriod.ONE_YEAR:
+          dateRange = pastDateRange("day", 365);
+          step = "day";
+          labelUnit = "day";
+          break;
+        default:
+          dateRange = pastDateRange("week");
+          step = "day";
+          labelUnit = "day";
+          break;
+      }
       break;
-    case TimePeriod.THREE_MONTHS:
-      dateRange = pastDateRange("day", 90);
+    case FilterType.Earnings: {
+      if (periodRange.earningPeriod?.date === undefined) {
+        throw new Error("Undefined earning period date");
+      }
+      dateRange = {
+        startDate: dayjs(periodRange.earningPeriod.date).toDate(),
+        endDate: dayjs(periodRange.earningPeriod.date).add(1, "month").toDate(),
+      };
       step = "day";
       labelUnit = "day";
       break;
-    case TimePeriod.MONTH:
-      dateRange = pastDateRange("day", 30);
+    }
+    case FilterType.DateRange: {
+      if (periodRange.dateRange === undefined || periodRange.dateRange.startDate > periodRange.dateRange.endDate) {
+        throw new Error("Invalid date range");
```

Comment:
Great that you're adding all this error handling in the functions. However, not seeing those errors being handled on the front end when you call this function.